include/pybind11/common.h

```diff
@@ -235,6 +235,23 @@ struct buffer_info {
         }
     }
 
+    buffer_info(buffer_info &&other){
+        (*this) = std::move(other);
+    }
+
+    buffer_info& operator=(buffer_info &&rhs){
+        ptr = rhs.ptr;
+        itemsize = rhs.itemsize;
+        size = rhs.size;
+        format = std::move(rhs.format);
+        ndim = rhs.ndim;
+        std::swap(shape, rhs.shape);
+        std::swap(strides, rhs.strides);
```

Comment:
This is very minor, but is there any reason to swap these two instead of just replacing with `shape = std::move(rhs.shape);` and likewise for `stride`?  (`view`/`ownview`, below, obviously need to be swapped).  The underlying implementation probably amounts to the same thing, but it seems unintuitive why these two get swapped when all the ones above don't.